library/x509.c

```diff
@@ -516,10 +529,35 @@ int mbedtls_x509_get_name( unsigned char **p, const unsigned char *end,
         cur->next = mbedtls_calloc( 1, sizeof( mbedtls_x509_name ) );
 
         if( cur->next == NULL )
-            return( MBEDTLS_ERR_X509_ALLOC_FAILED );
+        {
+            ret = MBEDTLS_ERR_X509_ALLOC_FAILED;
+            goto error;
+        }
 
         cur = cur->next;
     }
+
+error:
+    prev = NULL;
```

Comment:
I think it would be a good idea. If it's not too much trouble to do it in this PR, it would be great; the second-best scenario is we do it in a follow-up PR soon; otherwise we can open a ticket to track it but then there's a risk it will remain undone for a long time then.

Reply:
Hmm I thought that idea sounded familiar, I think it is already done in #6013. I suppose we should try and get that through and then I can rewrite this

Reply:
Having discuss that within the team, let's decouple this PR from #6013 and do not try to introduce a new API to factorize the code freeing the chain of names in this PR.